packages/react/src/components/Placeholder/ServerPlaceholder.tsx

```diff
@@ -0,0 +1,111 @@
+import { nonSerializedProps, PlaceholderProps } from './models';
+import { ComponentMap } from '../sharedTypes';
+import {
+  getComponentForRendering,
+  getPlaceholderRenderings,
+  getRenderedComponentProps,
+  renderEmptyPlaceholder,
+  wrapErrorBoundary,
+} from './PlaceholderCommon';
+import React from 'react';
+import { PlaceholderMetadata } from './PlaceholderMetadata';
+import { ComponentRendering } from '@sitecore-content-sdk/core/layout';
+
+export type ServerPlaceholderProps = PlaceholderProps & {
+  componentMap: ComponentMap;
+};
+
+export const ServerPlaceholder = (props: ServerPlaceholderProps) => {
+  if (!props.componentMap) {
```

Comment:
do we need this check? since componentMap is required

Reply:
made componentMap optional - we need to have it so for compatibilty between client/server placeholder